--- a/drivers/net/wireless/ath/ath10k/core.c
+++ b/drivers/net/wireless/ath/ath10k/core.c
@@ -3332,7 +3332,14 @@ static int ath10k_core_probe_fw(struct a
 
 	device_get_mac_address(ar->dev, ar->mac_addr, sizeof(ar->mac_addr));
 
+#if LINUX_VERSION_IS_LESS(5,10,0)
+	const char *mac;
+	mac = of_get_mac_address(ar->dev->of_node);
+	if (mac)
+		ether_addr_copy(ar->mac_addr, mac);
+#else
 	of_get_mac_address(ar->dev->of_node, ar->mac_addr);
+#endif
 
 	ret = ath10k_core_init_firmware_features(ar);
 	if (ret) {