--- a/drivers/net/wireless/ath/ath9k/init.c
+++ b/drivers/net/wireless/ath/ath9k/init.c
@@ -691,7 +691,14 @@ static int ath9k_of_init(struct ath_soft
 		ah->ah_flags |= AH_NO_EEP_SWAP;
 	}
 
+#if LINUX_VERSION_IS_LESS(5,10,0)
+	const char *mac;
+	mac = of_get_mac_address(np);
+	if (mac)
+		ether_addr_copy(common->macaddr, mac);
+#else
 	of_get_mac_address(np, common->macaddr);
+#endif
 
 	return 0;
 }